### PR TITLE
DynamiCertificateManager: certificate locked at WAITING state

### PR DESCRIPTION
**Pull request by Pols95**
* Added more debug
* Moved to FINE/DEBUG some too verbose INFO logs
* Fixed bug for which WAITING certificates were not saved on db whether AVAILABLE ones were processed before them
* Fixed a NPE in CertificatesResource when trying to stringify the certificate status (status is null as long as the certificate isn't stored to the db, which happens only when the manager successfully process it)
* Now certificates are processed according to the hostname in ascending alphabetically order

### CODE REVIEW

File: carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java [hunk 1]
```diff
@@ -325,7 +325,10 @@ private void certificatesLifecycle() {
                     groupMembershipHandler.fireEvent(EVENT_CERT_AVAIL_CHANGED);
                 }
             } catch (AcmeException | IOException | ConfigurationStoreException | GeneralSecurityException | IllegalStateException ex) {
+                LOG.log(Level.SEVERE, null, ex);
```
Comment (eolivelli) [inline carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java]: this is okay, we have to leave this line

File: carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java [hunk 2]
```diff
@@ -325,7 +325,10 @@ private void certificatesLifecycle() {
                     groupMembershipHandler.fireEvent(EVENT_CERT_AVAIL_CHANGED);
                 }
             } catch (AcmeException | IOException | ConfigurationStoreException | GeneralSecurityException | IllegalStateException ex) {
+                LOG.log(Level.SEVERE, null, ex);            
+            } catch (Throwable ex) {
                 LOG.log(Level.SEVERE, null, ex);
```
Comment (eolivelli) [inline carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java]: catch  Throwable is a code smell
and re-throwing it is also bad

File: carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java (hunk 1 above)
Comment (eolivelli) [reply carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java]: please also add a message like 'Error while handling dynamic certificate xxxx"....

File: carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java (hunk 2 above)
Comment (Pols95) [reply carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java]: yep but maybe is the only way to find out what happens... i'll remove it asap

File: carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java [hunk 3]
```diff
@@ -207,21 +209,26 @@ public void run() {
...
+            boolean updateDB = true;
+            boolean notifyCertAvailChanged = false;
             try {
                 CertificateData cert = loadOrCreateDynamicCertificateForDomain(domain);
                 switch (DynamicCertificateState.valueOf(cert.getState())) {
                     case WAITING: { // certificate waiting to be issues/renew
                         LOG.info("Certificate ISSUING process for domain: " + domain + " STARTED.");
                         Order order = client.createOrderForDomain(domain);
                         cert.setPendingOrderLocation(order.getLocation());
+                        LOG.fine("Pending order location for domain " + domain + ": " + order.getLocation());
```
Comment (eolivelli) [inline carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java]: you can leave at INFO, this is important and we are not ordering new certificate frequently

File: carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java [hunk 4]
```diff
@@ -207,21 +209,26 @@ public void run() {
...
                         Http01Challenge challenge = client.getHTTPChallengeForOrder(order);
                         if (challenge == null) {
+                            LOG.info("VERIFIED");
                             cert.setState(VERIFIED);
                         } else {
-                            triggerChallenge(challenge);
+                            LOG.fine("Pending challenge data for domain " + domain + ": " + challenge.getJSON());
```
Comment (eolivelli) [inline carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java]: you can leave at INFO, this is important and we are not ordering new certificate frequently

File: carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java [hunk 5]
```diff
@@ -207,21 +209,26 @@ public void run() {
...
                 CertificateData cert = loadOrCreateDynamicCertificateForDomain(domain);
                 switch (DynamicCertificateState.valueOf(cert.getState())) {
                     case WAITING: { // certificate waiting to be issues/renew
                         LOG.info("Certificate ISSUING process for domain: " + domain + " STARTED.");
                         Order order = client.createOrderForDomain(domain);
                         cert.setPendingOrderLocation(order.getLocation());
+                        LOG.fine("Pending order location for domain " + domain + ": " + order.getLocation());
                         Http01Challenge challenge = client.getHTTPChallengeForOrder(order);
                         if (challenge == null) {
+                            LOG.info("VERIFIED");
```
Comment (eolivelli) [inline carapace-server/src/main/java/org/carapaceproxy/server/certiticates/DynamicCertificatesManager.java]: without context this line is meaning less

**Comment by eolivelli** [comment]
@Pols95  I have added a commit to use "openjdk11" instead of "oraclejdk11" because it does not work anymore

**Pull request merged**